service/controller/key/key.go

```diff
@@ -137,13 +137,30 @@ func BucketName(customObject v1alpha1.AWSConfig, accountID string) string {
 	return fmt.Sprintf("%s-g8s-%s", accountID, ClusterID(customObject))
 }
 
-// BucketObjectName computes the S3 object path to the actual cloud config.
-//
-//     /version/3.4.0/cloudconfig/v_3_2_5/master
-//     /version/3.4.0/cloudconfig/v_3_2_5/worker
-//
-func BucketObjectName(customObject v1alpha1.AWSConfig, role string) string {
-	return fmt.Sprintf("version/%s/cloudconfig/%s/%s", VersionBundleVersion(customObject), CloudConfigVersion, role)
+// ReleaseComponentHash returns a value unique to the
+func ReleaseComponentHash(release releasev1alpha1.Release) (string, error) {
```

Comment:
This bad boy needs some unit tests. It looks correct but we should ensure that it is actually doing what we expect.

Reply:
I decided to hash the rendered ignition rather than the release so this is no longer relevant. The interesting logic is in `internal/detection` now.